controllers/main.go

```diff
@@ -119,43 +120,18 @@ func getClientIP(r *http.Request, realIPHeader string) string {
 }
 
 // NewMainController is the constructor for the entire controller routing.
-func NewMainController(params *chaincfg.Params, adminIPs []string,
-	adminUserIDs []string, APISecret string, APIVersionsSupported []int, baseURL string,
-	closePool bool, closePoolMsg string, feeKey *hdkeychain.ExtendedKey,
-	stakepooldConnMan *stakepooldclient.StakepooldManager, poolFees float64,
-	poolEmail, poolLink string, emailSender email.Sender, realIPHeader string,
-	voteKey *hdkeychain.ExtendedKey, maxVotedTickets int, description string,
-	designation string) (*MainController, error) {
+func NewMainController(mainController *MainController) (*MainController, error) {
```

Comment:
I think you should make a type Config struct {} holding the configuration options you want and then pass to NewMainController.

```go
type Config struct {
   ...
}

type MainController struct {
   ...
   cfg Config
}

func NewMainController(cfg *Config) (*MainController, error) {
```

Reply:
Ah, opposite of my suggestion. Sorry about that 😬

If I may ask for future reference, what's the logic behind that @dajohi ?
And, it's best to keep the cfg tag? You could embed it I think like 
`type MainController struct {
   *Config
   ...
   }`
Thanks.

Reply:
I don't mind the embed.  just really want a Config type.